src/Core/Services/Implementations/UserService.cs

```diff
@@ -521,6 +524,13 @@ public async Task<IdentityResult> ChangeEmailAsync(User user, string masterPassw
             return IdentityResult.Failed(_identityErrorDescriber.PasswordMismatch());
         }
 
+        var managedUserValidationResult = await ValidateManagedUserAsync(user, newEmail);
```

Comment:
I've tried to add test coverage for this method; however, it's using a method from UserManager, which is very difficult to mock. It looks like other devs have used some hacks to work around a similar issue, but I'm not sure if that approach will work in our case. This is also probably why there are no tests for this method.